chore: Use SQLAlchemy 2 and support `psycopg[binary]`

edgarrmondragon opened the pull request.

edgarrmondragon committed:
chore: Use `psycopg[binary]`

WillDaSilva commented:
This will supersede (unless we feel like updating it in `main` before Meltano v3 is released):
- #7546

edgarrmondragon committed:
Merge branch 'cloud' into chore/use-psycopg3

edgarrmondragon committed:
Use connection object in DB init hooks

edgarrmondragon committed:
Use `.mappings`

edgarrmondragon committed:
Migrate ORM usage

https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#migration-orm-usage

edgarrmondragon committed:
Ensure migrations are committed

edgarrmondragon committed:
Preserve support for psycopg2 driver

edgarrmondragon committed:
Preserve support for psycopg2 driver

edgarrmondragon committed:
Fix text execution in migration

edgarrmondragon committed:
Add TODO for mypy error

edgarrmondragon committed:
Attempt to fix plugin_settings.label type

edgarrmondragon committed:
Limit varchar size for mssql

edgarrmondragon committed:
Set charset

edgarrmondragon committed:
0.0.0.0 -> localhost

edgarrmondragon commented:
Having a few problems testing mssql locally: https://github.com/pymssql/pymssql/issues/822

WillDaSilva commented on pyproject.toml:
```diff
@@ -61,6 +61,7 @@ jsonschema = "^4.17"
 packaging = "^23.0"
 platformdirs = "^3.0.0"
 psutil = "^5.9.5"
+psycopg = {version = "^3.1.8", extras=["binary"]}
 psycopg2-binary = "^2.8.5"
```
We'll have to note this in the V2 -> V3 migration guide:
- https://github.com/meltano/meltano/issues/7957

edgarrmondragon committed:
Merge branch 'cloud' into chore/use-psycopg3

edgarrmondragon committed:
Make pgsql packages into extras

edgarrmondragon committed:
Use simple DROP DATABASE

edgarrmondragon committed:
Revert "Use simple DROP DATABASE"

This reverts commit 13e21bf622f37b240966f84ff02f944a1356812a.

edgarrmondragon committed:
Use `text` instead of `DDL`

edgarrmondragon commented:
I'm splitting this PR into 3 smaller changes to make it easier to debug what's causing MSSQL to fail:

- https://github.com/meltano/meltano/pull/7960
- Bump SQLAlchemy to 2.0
- Add new `postgres` extra and `psycopg[binary]` dependency

edgarrmondragon commented:
All items above have been addressed in their own PRs
